src/internal/nativeModule.web.ts

```diff
@@ -0,0 +1,286 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+import {
+  NetInfoNativeModule,
+  DEVICE_CONNECTIVITY_EVENT,
+  NetInfoNativeModuleState,
+} from './privateTypes';
+import {
+  NetInfoState,
+  NetInfoStateType,
+  NetInfoUnknownState,
+  NetInfoNoConnectionState,
+  NetInfoCellularState,
+  NetInfoBluetoothState,
+  NetInfoEthernetState,
+  NetInfoWifiState,
+  NetInfoWimaxState,
+  NetInfoOtherState,
+  NetInfoCellularGeneration,
+} from './types';
+
+// See https://wicg.github.io/netinfo/#dom-connectiontype
+type ConnectionType =
+  | 'bluetooth'
+  | 'cellular'
+  | 'ethernet'
+  | 'mixed'
+  | 'none'
+  | 'other'
+  | 'unknown'
+  | 'wifi'
+  | 'wimax';
+
+// See https://wicg.github.io/netinfo/#dom-effectiveconnectiontype
+type ConnectionEffectiveType = '2g' | '3g' | '4g' | 'slow-2g';
+
+// https://wicg.github.io/netinfo/#dom-networkinformation-savedata
+type ConnectionSaveData = boolean;
+
+interface Events {
+  change: Event;
+}
+
+interface Connection {
+  type: ConnectionType;
+  effectiveType: ConnectionEffectiveType;
+  saveData: ConnectionSaveData;
+  addEventListener<K extends keyof Events>(
+    type: K,
+    listener: (event: Events[K]) => void,
+  ): void;
+  removeEventListener<K extends keyof Events>(
+    type: K,
+    listener: (event: Events[K]) => void,
+  ): void;
+}
+
+// Create (optional) connection APIs on navigator
+declare global {
+  interface Navigator {
+    connection?: Connection;
+    mozConnection?: Connection;
+    webkitConnection?: Connection;
+  }
+}
+
+// Check if the browser supports the connection API
+const connection =
+  window.navigator.connection ||
```

Comment:
I advise you to check the implementation of netinfo (compatible with version 4.x) present in my library https://github.com/morrys/wora/blob/master/packages/netinfo/src/NetInfo.ts#L14 at order to support the SSR